### ISSUE TITLE
[!] Found packet with bad FCS, skipping...

### ISSUE THREAD

**Issue by Mi-Al** [v1s1t0r1sh3r3/airgeddon#54]
On Kali Linux I have permanent error I use Wash either Airgeddon:

```
[!] Found packet with bad FCS, skipping...
[!] Found packet with bad FCS, skipping...
[!] Found packet with bad FCS, skipping...
[!] Found packet with bad FCS, skipping...
[!] Found packet with bad FCS, skipping...
[!] Found packet with bad FCS, skipping...
```

Option **--ignore-fcs** or **-C** fixed it for me. Would you like to add the option to Wash in Airgeddon?

![03](https://cloud.githubusercontent.com/assets/21211053/21538438/23449ec2-cdae-11e6-91b4-5833af1f5421.jpg)

**Comment by v1s1t0r1sh3r3** [comment]
Quite weird... because airgeddon checks before scanning looking precisely for that error and if there is bad fcs it puts the "-C" parameter. The reason of this check instead to put it directly is because in other Linux like Wifislax, is the opposite, with -C you get the bad fcs and without the parameter it works... so I decided to perform a little test always to check this and then, airgeddon decide if the parameter -C is necessary.

What is your wireless card chipset? Is your problem only in Kali Linux?

**Comment by Mi-Al** [comment]
My Wi-Fi adapter is Alfa AWUS052NH with chipset Ralink Technology, Corp. RT3572, driver rt2800usb.
I also tested on Arch Linux, there I have no this error. Wash can see networks without the option.

**Comment by kcdtv** [comment]
Hi there! 
If you use the latest version of Reaver you do not need in any case the -C option.
Just download and update your Reaver version from the github repositories: [https://github.com/t6x/reaver-wps-fork-t6x](url)
That is something pending for 2017, i think that in a couple of weeks (or a bit more) I will contact the kali crew to update the version from their repositories. 
There are still some mini bugs and warnings to fix, some new option to polish, testing to be done but there should be soon a "stable release" that i hope will go in Kali repositories.
For the moment just update your reaver with the git hub version
It is stable enough, you will see, you will not lose efficiency . 
Bye!

**Comment by Mi-Al** [comment]
Oh, it helped, thanks!
Now I think The program had different behavior in Arch Linux because I installed reaver-wps-fork-t6x long time ago and just forgot. I was playing with WPS problem for a while and tested different forks.

**Comment by v1s1t0r1sh3r3** [comment]
So what is the point to implement on airgeddon? I can add the -C option only for certain versions. What is the phormula?

**Comment by v1s1t0r1sh3r3** [comment]
This is a weird thing... ok, let's fix this... look at this, is the code for the parametrization:

```
#Perform a test to determine if fcs parameter is needed on wash scanning
function set_wash_parametrization() {

	fcs=""
	declare -gA wash_ifaces_already_set
	readarray -t WASH_OUTPUT < <(timeout -s SIGTERM 1 wash -i "${interface}" 2> /dev/null)

	for item in "${WASH_OUTPUT[@]}"; do
		if [[ ${item} =~ ^\[\!\].*bad[[:space:]]FCS ]]; then
			fcs="-C"
			break
		fi
	done

	wash_ifaces_already_set[${interface}]=${fcs}
}
```
It launchs a wash command with a timeout of 1 second and parse the output. There are two tests you can do. I know you are advanced user :smile: so you can go through the code to search this function I showed here and modify it:
 - Change the timeout to 3 seconds for example. And lets see if now it has time to get the error and parse it.
 - Change the regexp to `if [[ ${item} =~ bad[[:space:]]FCS ]]; then` to see if could be some problem with some char.

Please, do the tests separately... and after each test, you must reboot airgeddon because as you know, it keeps for each session if an interface must have the -C option or not. Get back with the results please.

Thank you!

**Comment by Mi-Al** [comment]
I restore Reaver from Kali Linux reps and changed the timeout to 3 seconds – this helped! I can see APs and no errors.
The second way did not take an effect.
My hardware is ok, but I run Kali Linux in VirtualBox.

**Comment by v1s1t0r1sh3r3** [comment]
Ok, so the problem is the 1 second timeout right?... I'll update the script with some minor changes and this is one of them. I'll put 2 seconds. Is ok with 2 seconds? if you can test it could be nice. Thanks.

**Comment by Mi-Al** [comment]
Yep, timeout in 2 seconds also works.

**Comment by v1s1t0r1sh3r3** [comment]
Already patched. Now this fix (and more stuff) is on v5.14. You have it on dev branch. There are more translations there (only once, string 395) and you can proceed with it. This bug will be automatically closed when this comes to master.
